sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubConsumer.java

```diff
@@ -60,9 +74,43 @@
      *     batch; if not specified, the default wait time specified when the consumer was created will be used.
      * @return A set of {@link EventData} that was received. The iterable contains up to {@code maximumMessageCount}
      *     events.
+     * @throws NullPointerException if {@code maximumWaitTime} is null.
+     * @throws IllegalArgumentException if {@code maximumMessageCount} is less than 1 or {@code maximumWaitTime} is
+     *     zero or a negative duration.
      */
     public IterableStream<EventData> receive(int maximumMessageCount, Duration maximumWaitTime) {
-        return new IterableStream<>(Flux.empty());
+        Objects.requireNonNull(maximumWaitTime, "'maximumWaitTime' cannot be null.");
+
+        if (maximumMessageCount < 1) {
+            throw new IllegalArgumentException("'maximumMessageCount' cannot be less than 1.");
+        } else if (maximumWaitTime.isNegative() || maximumWaitTime.isZero()) {
+            throw new IllegalArgumentException("'maximumWaitTime' cannot be zero or less.");
```

Comment:
(extreme) nit:  Same error condition for these two exceptions, but different wording.   It may be helpful, if not superfluous, to use consistent verbiage.